src/Cli/dotnet/ParseResultExtensions.cs

```diff
@@ -33,8 +33,11 @@ public static void ShowHelpOrErrorIfAppropriate(this ParseResult parseResult)
         public static string RootSubCommandResult(this ParseResult parseResult)
         {
             return parseResult.RootCommandResult.Children?
-                .FirstOrDefault(c => c.Token() != null && c.Token().Type.Equals(TokenType.Command))?
-                .Symbol.Name ?? string.Empty;
+                .Select(c => c.Token() == null ?
+                    parseResult.FindResultFor(Parser.DotnetSubCommand)?.GetValueOrDefault<string>() :
+                    c.Token().Type.Equals(TokenType.Command) ? c.Symbol.Name : null)
+                .Where(subcommand => !string.IsNullOrEmpty(subcommand))
+                .FirstOrDefault() ?? string.Empty;
```

Comment:
```suggestion
                .FirstOrDefault(subcommand => !string.IsNullOrEmpty(subcommand)) ?? string.Empty;
```

Reply:
This might benefit a comment. Like what the input is and what you're trying to do.

Reply:
Agree it is too complicated now, You can extract methods and add few tests just for RootSubCommandResult method

Reply:
I'll include this in the PR to add test coverage for #15349 (since this also fixes that issue).